### PR TITLE
pg_upgrade: add regression DB back to the ICW smoke test

### PR DESCRIPTION
**Pull request by jchampio**
As part of the 9.3 pg_upgrade backport, the regression database was temporarily dropped from the upgrade smoke tests to get things working. This PR adds the regression database back to the ICW, filtering out all the tables that are known to not upgrade correctly.

The list is pretty long -- the new logic to drop all AO tables cuts the regression database in half -- but this should make it easier to parallelize pg_upgrade fixes.

attn: @Eulerizeit

### CODE REVIEW

File: contrib/pg_upgrade/test_gpdb_pre.sql [hunk 1]
```diff
@@ -3,12 +3,146 @@
...
-DROP DATABASE regression;
-DROP DATABASE dsp3;
-DROP DATABASE isolation2test;
+-- drop all AO tables
+\ir pre_drop_ao.sql
 
+DROP DATABASE IF EXISTS isolation2test;
+
+-- drop all AO tables
+\ir pre_drop_ao.sql
```
Comment (danielgustafsson) [inline contrib/pg_upgrade/test_gpdb_pre.sql]: There seems to be a `\c <database>` missing for this one?

File: contrib/pg_upgrade/test_gpdb_pre.sql [hunk 2]
```diff
@@ -3,12 +3,146 @@
...
+
+DROP DATABASE IF EXISTS pgutest;
```
Comment (danielgustafsson) [inline contrib/pg_upgrade/test_gpdb_pre.sql]: If this database suddenly starts to appear from ICW we won't cover what's in it with this `DROP`. I'd rather have the test fail on the duplicate name so we realize and can rename this one.

File: contrib/pg_upgrade/test_gpdb.sh [hunk 3]
```diff
@@ -263,7 +263,10 @@ fi
 
 # Run any pre-upgrade tasks to prep the cluster
 if [ -f "test_gpdb_pre.sql" ]; then
-	psql -f test_gpdb_pre.sql postgres
+	if ! psql -f test_gpdb_pre.sql -v ON_ERROR_STOP=1 postgres; then
+		echo "ERROR: unable to execute pre-upgrade cleanup"
+		exit 1
+	fi
```
Comment (danielgustafsson) [inline contrib/pg_upgrade/test_gpdb.sh]: This should probably be a separate commit as it's a good addition on its own.

File: contrib/pg_upgrade/test_gpdb_pre.sql (hunk 1 above)
Comment (jchampio) [reply contrib/pg_upgrade/test_gpdb_pre.sql]: Good catch! Will fix.

File: contrib/pg_upgrade/test_gpdb_pre.sql (hunk 2 above)
Comment (jchampio) [reply contrib/pg_upgrade/test_gpdb_pre.sql]: Hmm, okay. I added this piece because previously you could run `make check` over and over again, but with the new `ON_ERROR_STOP=1` it fails when creating the database a second time.

What's the goal of this `pgutest` database? Does it belong somewhere else now that we're adding `regression` back in?

File: contrib/pg_upgrade/test_gpdb.sh (hunk 3 above)
Comment (jchampio) [reply contrib/pg_upgrade/test_gpdb.sh]: Will do.

File: contrib/pg_upgrade/test_gpdb_pre.sql (hunk 2 above)
Comment (jchampio) [reply contrib/pg_upgrade/test_gpdb_pre.sql]: > Do you see a use for having a non-ICW database in the upgrade test path, or should we just rip it out?

I think it makes sense to remove it.

**Comment by danielgustafsson** [comment]
> On 31 Aug 2018, at 00:36, Jacob Champion <notifications@github.com> wrote:
> 
> @jchampio commented on this pull request.
> 
> In contrib/pg_upgrade/test_gpdb_pre.sql <https://github.com/greenplum-db/gpdb/pull/5621#discussion_r214201555>:
> 
> >  
> +DROP DATABASE IF EXISTS isolation2test;
> +
> +-- drop all AO tables
> +\ir pre_drop_ao.sql
> +
> +DROP DATABASE IF EXISTS pgutest;
> Do you see a use for having a non-ICW database in the upgrade test path, or should we just rip it out?
> 
> I think it makes sense to remove it.
> 
Sounds good to me, are you doing that in this PR? (FWIW, +1 on doing that here)

**Comment by jchampio** [comment]
@danielgustafsson Yep, I just pushed a fixup. We're working on pulling out that extra commit now.

File: contrib/pg_upgrade/test_gpdb_pre.sql (hunk 2 above)
Comment (danielgustafsson) [reply contrib/pg_upgrade/test_gpdb_pre.sql]: :+1: on your pushed change

**Comment by jchampio** [comment]
@danielgustafsson Thanks for the review!

**Comment by jchampio** [comment]
Rebased and rewritten; this is ready to go in once master goes greenish.
